pkg/operatorstatus/operator_status.go

```diff
@@ -214,3 +206,25 @@ func printOperandVersions(versions []configv1.OperandVersion) string {
 
 	return strings.Join(versionsOutput, ", ")
 }
+
+func clusterObjectNeedsUpdating(co *configv1.ClusterOperator, conds []configv1.ClusterOperatorStatusCondition, desiredVersions []configv1.OperandVersion, desiredRelatedObjects []configv1.ObjectReference) (*configv1.ClusterOperator, bool) {
```

Comment:
It looks like we only use the `.Status` field on `co`. Could we switch to passing that in?

Reply:
We might want to update the status based on the values on spec, I think it is worth keeping this way